src/main/java/org/openstreetmap/atlas/geography/atlas/BareAtlas.java

```diff
@@ -560,9 +568,24 @@ public SnappedEdge snapped(final Location point, final Distance threshold)
         nodesWithin(boundary, hasNode.negate()).forEach(
                 node -> builder.addNode(node.getIdentifier(), node.getLocation(), node.getTags()));
 
-        // Add the edges
-        edgesIntersecting(boundary).forEach(
-                edge -> builder.addEdge(edge.getIdentifier(), edge.asPolyLine(), edge.getTags()));
+        // Add the edges. Use a consumer that makes sure master edges are always added first.
+        final Consumer<Edge> edgeAdder = edge ->
+        {
+            if (builder.peek().edge(edge.getIdentifier()) == null)
+            {
+                if (edge.getIdentifier() != 0 && edge.hasReverseEdge())
```

Comment:
I agree.

It seems to me that swapping `edge.getIdentifier() != 0` with `!isMasterEdge()` would correct this code segment.

Reply:
We have a lot of unit tests that unfortunately create edges with id 0 using `@TestAtlas`. The whole idea of having a `+` and `-` for forward and reverse edges breaks down unfortunately for edge 0, and all the tests break. This might be worth creating an issue to track it, however the impact is pretty small. Thoughts?

Reply:
That makes sense, it's impossible to know that without you explaining - I was trying to wrap my head around why 0 was there - it's worth putting that as a comment here. Otherwise, is it possible to clean up tests now? If not, creating an issue like you said to clean up the test and update this piece of code to not rely on 0 might be worthwhile.

Reply:
Cleaning up the tests will be a full task. Created an issue #252, will add comments too.